File: solutions/fibonacci-number/fibonacci-number.py
// @Title: Fibonacci Number
// @Author: colinjxc
// @Date: 2022-02-05T02:44:37+08:00
// @URL: https://leetcode-cn.com/problems/fibonacci-number


class Solution:
    def fib(self, n: int) -> int:
        return int((5**0.5)*0.2*( ((1+5**0.5)/2)**n-((1-5**0.5)/2)**n))

File: solutions/fibonacci-number/fibonacci-number.go
// @Title: Fibonacci Number
// @Author: colinjxc
// @Date: 2022-02-05T02:28:51+08:00
// @URL: https://leetcode-cn.com/problems/fibonacci-number


func fib(n int) int {
    if n < 2 {
        return n
    }
    return fib(n-1) + fib(n-2)
}
